enhancements/olm-managed-metrics.md

```diff
@@ -0,0 +1,84 @@
+
+It is imparative that OLM provide these teams with the means to interact with Prometheus operators on cluster and by extension report metrics to Telemeter.
+
+### Goals
+
+* Provide operators with the means to expose metrics to prometheus via `ServiceMonitors`
+* Provide operators with the means to create `PrometheusRules`
+* Ensure that only approved Red Hat operators are scraped by Telemeter
+
+### Non-Goals
+
+* Provide generic metrics about OLM managed operators
+
+## Proposal
+
+### Bundle ServiceMonitor and PrometheusRule Objects with a CSV
+
+The bundle object should be updated to support Prometheus `ServiceMonitor` and `PrometheusRule` objects as it does with RBAC. When these objects are packaged with a CSV, OLM will be updated to deploy them along with the operator.
+
+### OLM Telemeter Support
+
+#### Telemeter Requirements
+
+Once OLM supports bundling `ServiceMonitor` and `PrometheusRule` objects with the CSV, OLM must enable Red Hat owned operators to take advantage of Telemeter. Before moving forward, let's review the requirements that must be met for Telemeter to scrape an application's metrics:
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labbeled with`openshift.io/cluster-monitoring=true`
+2. A [ServiceMonitor object](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources) must be created in the namespace mentioned above and point to the application's metrics endpoint
+3. A role/rolebinding must be created that provides the prometheus service account with appropriate RBAC priviledges to discover the ServiceMonitor object in the newly created namespace.
+
+Requirement 2 is fully met in [#Bundle ServiceMonitor/PrometheusRule Objects with a CSV](#bundle-serviceMonitor-prometheusRule-objects-with-a-csv), leaving our team with the duty to fulfill requirements 1 and 3.
+
+#### Allow operator author to define a specific namespace
+
+The first step towards allowing OLM managed operators to expose metrics to Telemeter involves deploying the operator in a namespace defined by the operator author. This can be accomplished by updating the `StrategyDetailsDeployment` type to include an optional field called `namespace` that defines where the operator's namespaced resources must be deployed. 
+
+#### Creating RBAC and applying the `openshift.io/cluster-monitoring=true` label
+
+We had considered creating any namespaces that begin with `openshift-` to include the `openshift.io/cluster-monitoring=true` label. However, this could be abused by external operators to report metrics to telemeter, potentially overloading the Telemeter prometheus instance. 
+
+To protect Telemeter, we propose creating a new operator that is responsible for adding the `openshift.io/cluster-monitoring=true` label to namespaces and for generating the Telemter RBAC. This operator would:
```

Comment:
Just a thought here, but what if this was defined by the catalog the operator came from (i.e. have something special about the registry for `RedHat` and `Certified` catalogs that tells you that you can use these metrics). That would probably be easier than trying to maintain a specific list of "approved" operators --> the approval just comes from being in the curated list.

Reply:
Suggestion added in the updated version.